engine/class_modules/paladin/sc_paladin.hpp

```diff
@@ -1126,28 +1127,39 @@ struct holy_power_consumer_t : public Base
     ab::execute();
 
     // if this is a vanq-hammer-based DS, don't do this stuff
-    if ( ab::background )
+    if ( ab::background && is_divine_storm )
       return;
 
     // Crusade and Relentless Inquisitor gain full stacks from free spells, but reduced stacks with FoJ
+    int num_stacks = as<int>( hp_used == 0 ? ab::base_costs[ RESOURCE_HOLY_POWER ] : hp_used );
+
+    if ( p -> azerite.relentless_inquisitor.ok() )
+      p -> buffs.relentless_inquisitor_azerite -> trigger( num_stacks );
+
+    if ( p -> legendary.relentless_inquisitor -> ok() && !is_vanq_hammer )
+      p -> buffs.relentless_inquisitor -> trigger();
+
     if ( p -> buffs.crusade -> check() )
     {
-      int num_stacks = as<int>( hp_used == 0 ? ab::base_costs[ RESOURCE_HOLY_POWER ] : hp_used );
       if ( p -> bugs && is_divine_storm && ( p -> buffs.empyrean_power_azerite -> up() || p -> buffs.empyrean_power -> up() ) )
-      {
-        num_stacks = 0;
-      }
+        ;
       else
-      {
         p -> buffs.crusade -> trigger( num_stacks );
```

Comment:
This should probably get reformulated.

Reply:
Not sure how to make it clearer other than just adding comments. Or replacing `;` with`{ }`. Ret thing I don't really want to touch it

Reply:
You could just reverse the condition so instead of having if (...) {} else x, you'd have if ( ! ... ) { x }

Reply:
Annoying to read tho. But sure